Source/WebGPU/WGSL/Metal/MetalFunctionWriter.cpp

```diff
@@ -494,7 +494,7 @@ void FunctionDefinitionWriter::visit(const Type* type)
         },
         [&](const Vector& vector) {
             auto* primitive = std::get_if<Primitive>(vector.element);
-            if (primitive && m_structRole.has_value() && *m_structRole == AST::StructureRole::PackedResource) {
+            if (primitive && vector.size == 3 && m_structRole.has_value() && *m_structRole == AST::StructureRole::PackedResource) {
```

Comment:
this looks like a good opportunity to use a function with a descriptive name, to help readers understand why "3" is special.